src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/service/cas2bail/Cas2BailApplicationNoteService.kt

```diff
@@ -0,0 +1,171 @@
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.results.ValidatableActionResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.EmailNotificationService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.ExternalUserService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.HttpAuthService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.NomisUserService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.service.cas2.UserAccessService
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.toCas2UiFormat
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.toCas2UiFormattedHourOfDay
+import java.time.OffsetDateTime
+import java.util.*
+
+@Service("Cas2BailApplicationNoteService")
+class Cas2BailApplicationNoteService(
+  private val cas2BailApplicationRepository: Cas2BailApplicationRepository,
+  private val cas2BailAssessmentRepository: Cas2BailAssessmentRepository,
+  private val cas2BailApplicationNoteRepository: Cas2BailApplicationNoteRepository,
+  private val userService: NomisUserService,
+  private val externalUserService: ExternalUserService,
+  private val httpAuthService: HttpAuthService,
+  private val emailNotificationService: EmailNotificationService,
+  private val userAccessService: UserAccessService,
+  private val notifyConfig: NotifyConfig,
+  @Value("\${url-templates.frontend.cas2.application-overview}") private val applicationUrlTemplate: String,
+  @Value("\${url-templates.frontend.cas2.submitted-application-overview}") private val assessmentUrlTemplate: String,
+) {
+
+  private val log = LoggerFactory.getLogger(this::class.java)
+
+  @Suppress("ReturnCount")
+  fun createAssessmentNote(assessmentId: UUID, note: NewCas2ApplicationNote): CasResult<Cas2BailApplicationNoteEntity> {
+    val assessment = cas2BailAssessmentRepository.findByIdOrNull(assessmentId)
+      ?: return CasResult.NotFound()
+
+    val application = cas2BailApplicationRepository.findByIdOrNull(assessment.application.id)
+      ?: return CasResult.NotFound()
+
+    if (application.submittedAt == null) {
+      return CasResult.GeneralValidationError("This application has not been submitted")
+    }
+
+    val isExternalUser = httpAuthService.getCas2AuthenticatedPrincipalOrThrow().isExternalUser()
+    val user = getCas2User(isExternalUser)
+
+    if (!isExternalUser && !nomisUserCanAddNote(application, user as NomisUserEntity)) {
+      return CasResult.Unauthorised()
+    }
+
+    val savedNote = saveNote(application, assessment, note.note, user)
+
+    sendEmail(isExternalUser, application, savedNote)
+
+    return CasResult.Success( savedNote )
+  }
+
+  private fun sendEmail(
+    isExternalUser: Boolean,
+    application: Cas2BailApplicationEntity,
+    savedNote: Cas2BailApplicationNoteEntity,
+  ) {
+    if (isExternalUser) {
+      sendEmailToReferrer(application, savedNote)
+    } else {
+      sendEmailToAssessors(application, savedNote)
```

Comment:
Note that any email sent will be using the cas2 reply address. I guess using the same address for cas2bail is fine?